### ISSUE TITLE
Infix operators should stay the same v2

### ISSUE THREAD

**Issue by mtomko** [olafurpg/scalafmt#284]
Thanks for putting this tool together. I noticed this quirk and wasn't sure if it was intentional:

Original:
```scala
  private[reports] def headerText(colBarcodes: Seq[String], compat: Boolean): String = {
    s"Sequence\tTotal\t${colBarcodes.mkString("\t")}\t" +
    (if (compat) "Construct IDs" else "Potential IDs")
  }
```

Scalafmt:
```scala
  private[reports] def headerText(colBarcodes: Seq[String], compat: Boolean): String = {
    s"Sequence\tTotal\t${colBarcodes.mkString("\t")}\t" + (if (compat) "Construct IDs" else "Potential IDs")
  }
```

I'm not sure my original formatting is the ideal solution in this case; however it seems that scalafmt should respect the `--maxColumns` setting rather than removing the line break after the `+` operator.
```scala
  private[reports] def headerText(colBarcodes: Seq[String], compat: Boolean): String = {
    s"Sequence\tTotal\t${colBarcodes.mkString("\t")}\t" +
    (if (compat) "Construct IDs" else "Potential IDs")
  }
```

Using:
* 0.2.5
* --style default

I ran the transformation using the IntelliJ plugin, but I doubt that's relevant. My .scalafmt:
```
-s default
--allowNewlineBeforeColonInMassiveReturnTypes false
--maxColumn 120
```

**Comment by olafurpg** [comment]
First of all, scalafmt's output seems to fit the 120 character limit:

```scala
// Column 120                                                                                                          |
object A {
  private[reports] def headerText(colBarcodes: Seq[String], compat: Boolean): String = {
    s"Sequence\tTotal\t${colBarcodes.mkString("\t")}\t" + (if (compat) "Construct IDs" else "Potential IDs")
  }
}
```

However, I agree that it would be nice if scalafmt would keep the newline after the `+`. I've struggled a lot with infix applications. I'm starting to lean towards preserving whatever newlines/spaces the original source puts around infix operators, then scalafmt would keep the newline after the `+` in this example.

**Comment by mtomko** [comment]
Oh, doh, I had my IDE misconfigured. I should have measured before reporting. I agree that there may be some question of whether you should preserve the newline, but given that this does fit within the specified `--maxColumn` this is an inappropriate example.

To expound on the "preserve newlines" argument, I guess you could say that  when a programmer chooses to break a line on an infix operator, perhaps that's implying some semantic reason for doing so. That is, the programmer is suggesting a sensible break where the code before that operator makes sense by itself, and the code after that operator makes sense by itself. So maybe there's no need to *eliminate* existing line breaks, but I definitely agree that scalafmt should consider *introducing* them when lines exceed the configured maximum width.

Thanks!

**Comment by mtomko** [comment]
Interesting. I changed the `--maxColumn` setting to 100 and scalafmt gave me:

```scala
  private[reports] def headerText(colBarcodes: Seq[String], compat: Boolean): String = {
    s"Sequence\tTotal\t${colBarcodes.mkString("\t")}\t" + (if (compat) "Construct IDs"
                                                           else "Potential IDs")
  }
```
In this case, I *definitely* prefer my prior formatting because it seems that the infix operator should bind more loosely than the parenthesized expression.

**Comment by olafurpg** [comment]
>  it seems that the infix operator should bind more loosely than the parenthesized expression.

I agree. The thing is that infix operators turned out to be trickier than I anticipated, so I haven't been succesful at modeling such ideas. In 0.2.6 I changed the rules to take whatever the user provided, except when the operator is one of `+ - && ||` and the right hand side starts with `(`. That's a silly exception that's causing problems in your example. I can remove it in the next release.
